### PR TITLE
Move to real classes for i18n classes for proper typing in strict mode

### PR DESCRIPTION
**Pull request by benoit74**
This PR is preparatory work for #212. 

It is also a rework of #151 / #183 where I obviously was a bit too lazy.

Since this is ready and "significant" and isolated change, I submit it in a dedicated PR.

I've removed two tests which do not really make sense once we use real classes, and are not possible since we cannot init the class with dict values anymore.

CHANGELOG is not updated since anyway we will have to tackle this separately for 5.0.0

This has been tested locally for proper operation with pyright strict mode.

### CODE REVIEW

**Review by rgaudin — APPROVED** [review]
Thank you ; it looks good (but we need the docstrings!).

Unfortunately, looking at this makes me want to change it all. I find the API unintuitive and I dont understand why there's `Lang`, `LangAndDetails` and all those functions that looks low-level.

Actually, I've tried to reorganize it into something more user-friendly and will push one commit. You let me know what you think about it. It works (updated the tests) but we might want to reorganize the code as everything happens in a single method…

File: src/zimscraperlib/i18n.py [hunk 1]
```diff
@@ -16,66 +13,68 @@ class NotFoundError(ValueError):
...
+            and self.iso_639_2b == value.iso_639_2b
+            and self.iso_639_2t == value.iso_639_2t
+            and self.iso_639_3 == value.iso_639_3
+            and self.iso_639_5 == value.iso_639_5
+            and self.english == value.english
+            and self.native == value.native
+        )
+
+    def __repr__(self) -> str:
+        return (
```
[the remark this answers is not in the document]
Comment (benoit74) [reply src/zimscraperlib/i18n.py]: Indeed

File: src/zimscraperlib/i18n.py [hunk 2]
```diff
@@ -16,66 +13,68 @@ class NotFoundError(ValueError):
...
-    @property
-    def query(self) -> str:
-        """Query issued for these language details"""
-        return self["query"]
-
-    @property
-    def querytype(self) -> str:
-        """Type of query issued to retrieve language details"""
-        return self["querytype"]
+class Lang:
```
[the remark this answers is not in the document]
Comment (benoit74) [reply src/zimscraperlib/i18n.py]: Agreed

**Comment by benoit74** [comment]
I like the new API. I didn't wanted to totally break the old one because I (badly) assumed it was done on-purpose. But I agree it was a nightmare to read and maintain, and probably also to use. I will ask next time ^^

I add missing tests to have 100% coverage.

I propose to not mind about the fact that method is long / complex, I find it way easier to understand what we do (and hence to maintain) than what we had before or what we could have by splitting the code in multiple functions.

**Pull request merged**